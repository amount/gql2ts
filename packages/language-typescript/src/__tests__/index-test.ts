import {
  interfaceExtendListToString,
  DEFAULT_NAME_QUERY,
  DEFAULT_FORMAT_INPUT,
  DEFAULT_TYPE_PRINTER,
  DEFAULT_DOCUMENTATION_GENERATOR
} from '../';

describe('language-typescript', () => {
  describe('interfaceExtendListToString', () => {
    it('with non-empty array', () => {
      expect(
        interfaceExtendListToString(['test', 'test2', 'test3'])
      ).toMatchSnapshot();
    });
    it('with empty array', () => {
      expect(interfaceExtendListToString([])).toMatchSnapshot();
    });
  });

  describe('DEFAULT_NAME_QUERY', () => {
    it('anonymous', () => {
      expect(DEFAULT_NAME_QUERY({} as any)).toMatchSnapshot();
    });
    it('with name', () => {
      expect(
        DEFAULT_NAME_QUERY({ name: { value: 'test' } } as any)
      ).toMatchSnapshot();
    });
  });

  describe('DEFAULT_FORMAT_INPUT', () => {
    it('non optional', () => {
      expect(DEFAULT_FORMAT_INPUT('test', false, 'test2')).toMatchSnapshot();
    });
    it('optional', () => {
      expect(DEFAULT_FORMAT_INPUT('test', true, 'test2')).toMatchSnapshot();
    });
  });

  describe('DEFAULT_TYPE_PRINTER', () => {
    it('non null', () => {
      expect(DEFAULT_TYPE_PRINTER('test', false)).toMatchSnapshot();
    });
    it('null', () => {
      expect(DEFAULT_TYPE_PRINTER('test', true)).toMatchSnapshot();
    });
  });

  describe('DEFAULT_DOCUMENTATION_GENERATOR', () => {
    describe('with a description', () => {
      it('without tags', () => {
        expect(DEFAULT_DOCUMENTATION_GENERATOR({
          description: 'This is a thing',
          tags: []
        })).toMatchSnapshot();
      });
      it('with 1 tag', () => {
        expect(DEFAULT_DOCUMENTATION_GENERATOR({
          description: 'This is a thing',
          tags: [{ tag: 'default', value: 'myDefaultValue' }]
        })).toMatchSnapshot();
      });
      it('with >1 tag', () => {
        expect(DEFAULT_DOCUMENTATION_GENERATOR({
          description: 'This is a thing',
          tags: [
            { tag: 'default', value: 'myDefaultValue' },
            { tag: 'deprecated', value: 'Use the other field instead' }
          ]
        })).toMatchSnapshot();
      });
    });

    describe('without a description', () => {
      it('without tags', () => {
        expect(DEFAULT_DOCUMENTATION_GENERATOR({
          tags: []
        })).toMatchSnapshot();
      });
      it('with 1 tag', () => {
        expect(DEFAULT_DOCUMENTATION_GENERATOR({
          tags: [{ tag: 'default', value: 'myDefaultValue' }]
        })).toMatchSnapshot();
      });
      it('with >1 tag', () => {
        expect(DEFAULT_DOCUMENTATION_GENERATOR({
          tags: [
            { tag: 'default', value: 'myDefaultValue' },
            { tag: 'deprecated', value: 'Use the other field instead' }
          ]
        })).toMatchSnapshot();
      });
    });
  });
});
