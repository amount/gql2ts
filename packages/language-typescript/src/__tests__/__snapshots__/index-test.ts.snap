// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`language-typescript DEFAULT_DOCUMENTATION_GENERATOR with a description with >1 tag 1`] = `
"
  /**
   * This is a thing
   * @default \\"myDefaultValue\\"
   * @deprecated \\"Use the other field instead\\"
   */"
`;

exports[`language-typescript DEFAULT_DOCUMENTATION_GENERATOR with a description with 1 json tag 1`] = `
"
  /**
   * This is a thing
   * @default {\\"myDefault\\":\\"Value\\"}
   */"
`;

exports[`language-typescript DEFAULT_DOCUMENTATION_GENERATOR with a description with 1 tag 1`] = `
"
  /**
   * This is a thing
   * @default \\"myDefaultValue\\"
   */"
`;

exports[`language-typescript DEFAULT_DOCUMENTATION_GENERATOR with a description without tags 1`] = `
"
  /**
   * This is a thing
   */"
`;

exports[`language-typescript DEFAULT_DOCUMENTATION_GENERATOR without a description with >1 tag 1`] = `
"
  /**
   * @default \\"myDefaultValue\\"
   * @deprecated \\"Use the other field instead\\"
   */"
`;

exports[`language-typescript DEFAULT_DOCUMENTATION_GENERATOR without a description with 1 json tag 1`] = `
"
  /**
   * @default {\\"myDefault\\":\\"Value\\"}
   */"
`;

exports[`language-typescript DEFAULT_DOCUMENTATION_GENERATOR without a description with 1 tag 1`] = `
"
  /**
   * @default \\"myDefaultValue\\"
   */"
`;

exports[`language-typescript DEFAULT_DOCUMENTATION_GENERATOR without a description without tags 1`] = `""`;

exports[`language-typescript DEFAULT_ENUM_FORMATTER w/ deprecated value 1`] = `
"{
  
  /**
   * value A
   * @deprecated \\"Bad\\"
   */
a = 'a',

  /**
   * @deprecated \\"Bad\\"
   */
b = 'b',

  /**
   * value C
   */
c = 'c',
d = 'd',
e = 'e'
}"
`;

exports[`language-typescript DEFAULT_ENUM_FORMATTER w/ description 1`] = `
"{
  
  /**
   * value A
   */
a = 'a',
b = 'b'
}"
`;

exports[`language-typescript DEFAULT_ENUM_FORMATTER w/o description 1`] = `
"{
  a = 'a',
b = 'b'
}"
`;

exports[`language-typescript DEFAULT_ENUM_TYPE_BUILDER formats properly 1`] = `
"const enum Test {
  a = 'a',
b = 'b'
}"
`;

exports[`language-typescript DEFAULT_FORMAT_INPUT non optional 1`] = `"test: test2;"`;

exports[`language-typescript DEFAULT_FORMAT_INPUT optional 1`] = `"test?: test2;"`;

exports[`language-typescript DEFAULT_NAME_QUERY anonymous 1`] = `"Anonymous"`;

exports[`language-typescript DEFAULT_NAME_QUERY with name 1`] = `"Test"`;

exports[`language-typescript DEFAULT_TYPE_PRINTER non null 1`] = `"test | null"`;

exports[`language-typescript DEFAULT_TYPE_PRINTER null 1`] = `"test"`;

exports[`language-typescript interfaceExtendListToString with empty array 1`] = `""`;

exports[`language-typescript interfaceExtendListToString with non-empty array 1`] = `" extends test, test2, test3"`;
